llzk_backend/src/codegen.rs

```diff
@@ -76,39 +170,803 @@ impl<'ast, 'llzk> LlzkCodegen<'ast, 'llzk> {
         println!("{} {}", Color::Green.paint("Written successfully:"), filename);
         Ok(())
     }
+
+    /// Write the generated `Module` to a stderr (for debugging). If there is some verification
+    /// error, it will be printed in MLIR generic format rather than the LLZK IR assembly format.
+    fn write_to_stderr(self) {
+        unsafe extern "C" fn callback(s: mlir_sys::MlirStringRef, _user: *mut c_void) {
+            // SAFETY: MLIR promises s.data points to s.length bytes valid for the call duration.
+            let bytes = slice::from_raw_parts(s.data as *const u8, s.length);
+            let _ = std::io::stderr().lock().write_all(bytes);
+        }
+
+        unsafe {
+            mlir_sys::mlirOperationPrint(
+                self.module.as_operation().to_raw(),
+                Some(callback),
+                std::ptr::null_mut(),
+            );
+        }
+    }
+}
+
+/// Generate a `felt.const` operation from a BigInt. Returns an `Err` result if unsuccessful
+/// or if the number of bits required to represent the BigInt does not fit in 32 bits.
+fn new_felt_const_op<'llzk>(
+    codegen: &LlzkCodegen<'_, 'llzk>,
+    meta: &Meta,
+    from: &BigInt,
+) -> Result<Operation<'llzk>> {
+    // ASSERT: The circom parser always produces non-negative constants. These can be negated via
+    // PrefixOp but negative BigInt constants are never created directly.
+    assert_ne!(from.sign(), num_bigint_dig::Sign::Minus, "Felt constants must be non-negative");
+    let attr = FeltConstAttribute::parse(
+        codegen.context,
+        // use required bits +1 to ensure unsigned representation
+        u32::try_from(from.bits())? + 1,
+        from.to_string().as_str(),
+    );
```

Comment:
Questions:
1. Can a `BigInt` be casted into a `BigUint`? It may implement `TryInto`, not sure from the top of my head.
2. Did you enable the `bigint` feature in the `llzk` dependency?

If the answer for both is yes you don't need to use `parse` since there's another constructor that takes a `BigUint`

Reply:
The problem is, circom uses the `num_bigint_dig` crate but llzk-rs uses `num_bigint`.